pytest_splinter/plugin.py

```diff
@@ -398,6 +452,12 @@ def prepare_browser(request, parent):
                 request.addfinalizer(browser.quit)
         elif not browser:
             browser = browser_pool[browser_key] = get_browser(splinter_webdriver)
+        request.addfinalizer(functools.partial(
```

Comment:
so for session scoped browsers, this fails due to the `pytest_runtest_makereport` hook doing nothing useful (as it's session's request and not node's request)